views/partials/trial-search-result.html

```diff
@@ -2,21 +2,29 @@
   <h1 class="title"><a href="/trials/{{ trial.id }}">{{ trial.public_title }}</a></h1>
   <dl class="status meta">
 
+    {% set unknownValue = 'unknown' %}
     <dt title="Registration date" class="date"><span>Registration date:</span></dt>
     <dd>
-      <time datetime="{{ trial.registration_date }}">
-        {{ trial.registration_date | formatDate }}
-      </time>
+      {% if trial.registration_date %}
+        <time datetime="{{ trial.registration_date }}">
+          {{ trial.registration_date | formatDate }}
+        </time>
+      {% else %}
+        {{ unknownValue | underscoresToCapitalized }}
+      {% endif %}
     </dd>
 
-    {% set status = trial.status or 'unknown' %}
+    {% set status = trial.status or unknownValue %}
     <dt title="Status" class="progress"><span>Status:</span></dt>
     <dd>{{ status | underscoresToCapitalized }}</dd>
 
-    {% set recruitment_status = trial.recruitment_status or 'unknown' %}
+    {% set recruitment_status = trial.recruitment_status or unknownValue %}
     <dt title="Recruitment status" class="recruitment"><span>Recruitment status:</span></dt>
     <dd>{{ recruitment_status | underscoresToCapitalized }}</dd>
 
+    {% set source_name = trial.source.name or (unknownValue | underscoresToCapitalized) %}
```

Comment:
Feels like it should be `(trial.source.name | underscoresToCapitalized) or unknownValue`, no? (Considering `underscoresToCapitalized` returns `undefined` on `undefined` values